rosbag2_storage/src/rosbag2_storage/metadata_io.cpp

```diff
@@ -73,14 +93,42 @@ struct convert<rosbag2_storage::TopicInformation>
     return node;
   }
 
-  static bool decode(const Node & node, rosbag2_storage::TopicInformation & metadata)
+  static bool decode(const Node & node, rosbag2_storage::TopicInformation & metadata, int version)
   {
-    metadata.topic_metadata = node["topic_metadata"].as<rosbag2_storage::TopicMetadata>();
+    metadata.topic_metadata = versioned_decode<rosbag2_storage::TopicMetadata>(
+      node["topic_metadata"], version);
     metadata.message_count = node["message_count"].as<uint64_t>();
     return true;
   }
 };
 
+template<>
+struct convert<std::vector<rosbag2_storage::TopicInformation>>
+{
+  static Node encode(const std::vector<rosbag2_storage::TopicInformation> & rhs)
+  {
+    Node node(NodeType::Sequence);
+    for (auto value : rhs) {
+      node.push_back(value);
+    }
+    return node;
+  }
+
+  static bool decode(
+    const Node & node, std::vector<rosbag2_storage::TopicInformation> & rhs, int version)
+  {
+    if (!node.IsSequence()) {
+      return false;
```

Comment:
if we return false, do we abort? do we have any testing outside the happy path for this? I'd like to simplify this problem by just throwing and not catching if we can live with that for now. It's better than having libyaml catch this and hand us half-correct structures.

Reply:
Ah, good catch. When I first wrote this I specialized a `yaml-cpp` class that was not called directly and throws when convert returns false. I'll change it to throw that exception

Reply:
Oh wait, never mind. This is not the line i thought it was. These return values are checked to throw a `TypedBadConversion` in the yaml-cpp implementation. We don't catch the exception, it will halt the program (correctly)